app/src/main/java/com/nutomic/syncthingandroid/service/SyncthingRunnable.java

```diff
@@ -133,13 +133,13 @@ public String run(boolean returnStdOut) {
             Thread lWarn = null;
             if (returnStdOut) {
                 try {
-                    InputStreamReader isr = new InputStreamReader(process.getInputStream(), Charsets.UTF_8);
-                    BufferedReader br = new BufferedReader(isr);
+                    BufferedReader br = new BufferedReader(new InputStreamReader(process.getInputStream(), Charsets.UTF_8));
                     String line;
                     while ((line = br.readLine()) != null) {
                         Log.println(Log.INFO, TAG_NATIVE, line);
                         capturedStdOut = capturedStdOut + line + "\n";
                     }
+                    br.close();
```

Comment:
Probably want to do this in a finally, in case something throws.

Reply:
Yeah, got a logcat and indeed, after the "generatekeys" execution, the original code before this PR throws a lot of bufferedReader exceptions before the thread is terminated.

Reply:
Huh I've got a strange compiler warning when I want to add the br.close into the finally-block of function thread "log" (line 354). It complains an IOException has to be caught, so please don't wonder why there a two tries now to clean this up.

Reply:
Verified the lastest commit https://github.com/syncthing/syncthing-android/pull/1074/commits/f436d4ee58c40124f80a763449a0dcca730d70e6 working correctly. And hey, you're idea kicke ass - the warning lines because of bufferedReader no longer occur.